code/frameworks/vue3-vite/src/plugins/vue-template.ts

```diff
@@ -1,6 +1,6 @@
 import type { Plugin } from 'vite';
 
-export async function templateCompilation() {
+export async function templateCompilation(): Promise<Plugin> {
```

Comment:
I actually don't see a reason for this to be a promise in the first place cc @yannbf ?

Reply:
There is no reason. The others were needed because they do dynamic imports, but this one seems simple enough